test/replication-luatest/gh_10506_spam_in_netbox_logs_test.lua

```diff
@@ -0,0 +1,153 @@
+end
+
+g.test_no_spam_in_logs_while_connecting_to_non_existent_instance = function()
+    g.router:exec(function(reconnect_after)
+        local netbox_f = require("fiber").create(function()
+            require("net.box").connect("localhost:0",
+                                       {reconnect_after = reconnect_after})
+        end)
+        rawset(_G, "netbox_f", netbox_f)
+    end, {g.reconnect_after})
+    test_only_one_record_appears_in_logs(g.router, g.connection_refused_pattern,
+                                        g.reconnect_after * 2)
+    g.router:exec(function() _G.netbox_f:cancel() end)
+end
+
+g.test_no_spam_in_logs_when_connection_cannot_be_restored_anymore = function()
+    g.storage:update_box_cfg({listen = "localhost:0"})
+    local storage_uri = g.storage:exec(function()
+        return box.info.listen[1]
+    end)
+    g.router:exec(function(storage_uri, reconnect_after)
+        local net_box = require("net.box")
+        local conn = net_box.connect(storage_uri,
+                                     {reconnect_after = reconnect_after})
+        t.assert(conn:is_connected())
+    end, {storage_uri, g.reconnect_after})
+    g.storage:stop()
+    test_only_one_record_appears_in_logs(g.router, g.connection_refused_pattern,
+                                        g.reconnect_after * 2)
+    g.storage:start()
+end
+
+g.test_no_spam_in_logs_during_complex_connection_scenarios = function()
+    -- Since connection doesn't die immediately after a test, it may happen,
+    -- that previous tests affect the current one, which causes the flakiness.
+    -- We need to restart our router in order to ensure that there are no
+    -- connections.
+    g.router:restart()
+    g.storage:update_box_cfg({listen = "localhost:0"})
+    local storage_uri = g.storage:exec(function()
+        return box.info.listen[1]
+    end)
+    g.storage:stop()
+    -- Try to connect to non-existent storage. Should be only one
+    -- "Connection refused" log.
+    assert_socket_state(storage_uri, "closed")
+    g.router:exec(function(reconnect_after, storage_uri)
+        local netbox_f = require("fiber").create(function()
+            require("net.box").connect(storage_uri,
+                                       {reconnect_after = reconnect_after})
+        end)
+        rawset(_G, "netbox_f", netbox_f)
+    end, {g.reconnect_after, storage_uri})
+    test_only_one_record_appears_in_logs(g.router, g.connection_refused_pattern,
+                                         g.reconnect_after * 2)
+    -- Restart our storage and bind the previous uri to it. Should not be any
+    -- "Connection refused" logs. We check this by comparing "Connection refu
+    -- sed" logs with datetimes before and after our storage reconfiguring.
+    g.storage:start()
+    local conn_refused_log_before_restart =
+        g.router:grep_log(g.connection_refused_pattern)
+    g.storage:update_box_cfg({listen = storage_uri})
+    g.storage:grep_log("entering the event loop")
```

Comment:
You just grep it and do nothing, let's either wait for it, or drop, if it's not needed (it looks, we don't need it)